src/presign/participant.rs

```diff
@@ -753,13 +952,16 @@ impl PresignKeyShareAndInfo {
                 &crate::zkp::pienc::PiEncSecret::new(&k, &rho),
             )?;
             let r1_public = RoundOnePublic {
-                K: PaillierCiphertext(K.clone()),
-                G: PaillierCiphertext(G.clone()),
                 proof: proof.clone(),
             };
-            let _ = ret_publics.insert(*id, r1_public);
+            let _ = r1_publics.insert(*id, r1_public);
         }
 
+        let r1_public_broadcast = RoundOnePublicBroadcast {
+            K: PaillierCiphertext(K.clone()),
+            G: PaillierCiphertext(G.clone()),
+        };
```

Comment:
Would it make sense to add a constructor for `RoundOnePublicBroadcast`? I know this is small, but the visual noise adds up over time.

Reply:
The noise will get reduced with #21. If K and G are already `PaillierCiphertext`s, the constructor should be reduced to just cloning:
```
let r1_public_broadcast = RoundOnePublicBroadcast {
    K: K.clone(),
    G: G.clone(),
}
```